src/components/sign/Sign.js

```diff
@@ -20,10 +20,25 @@ class Sign extends Component {
     handleDeny = e => {
         e.preventDefault();
         Mixpanel.track("SIGN Deny the transaction")
-        const { callbackUrl, meta } = this.props;
+        const { callbackUrl, meta, signTxStatus } = this.props;
         // TODO: Dispatch action for app redirect?
         if (this.props.callbackUrl) {
-            window.location.href = addQueryParams(callbackUrl, { meta, errorCode: 'userRejected' })
+
+            if (signTxStatus?.success !== false) {
+                window.location.href = addQueryParams(callbackUrl, {
+                    meta,
+                    errorType: 'userRejected',
+                    errorMessage: 'User rejected transaction'
```

Comment:
Sorry about the delayed response here!  Just a re-cap of our discussion in Discord
1. If `errorMessage` is un-bounded and can contain large strings, this could break URL size limits, so we probably don't want to add it unless we are very sure of its content in all cases :)

2. We should keep the code argument as `errorCode`, so that app developers integrations that check for failures don't suddenly stop getting feedback for errors in the `errorCode` query param -- this would effectively be a breaking change for apps that do check for errors in this step, since the _same_ data would be available in a different query param than it used to be.

Reply:
Or we could truncate `errorMessage` if it's particularly long :). For backwards compat we want to ensure URLs are < 2048 characters.

Reply:
@MaximusHaximus, made the following changes: https://github.com/near/near-wallet/pull/1703/commits/d3a035dde558e39dddd1fd9c6a0b2e9d2c1cc6a9

1. Keep argument as `errorCode`
2. Wrap strings in `encodeURIComponent`
3. Limit `errorMessage` to max 100 characters